auth0/provider.go

```diff
@@ -23,6 +25,18 @@ func Provider() *schema.Provider {
 				Required:    true,
 				DefaultFunc: schema.EnvDefaultFunc("AUTH0_CLIENT_SECRET", nil),
 			},
+			"debug": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				DefaultFunc: func() (interface{}, error) {
```

Comment:
Can't just use `schema.EnvDefaultFunc` here because it returns the variable value as a string, but we need bool.

Reply:
Perhaps we don't need to use env vars for this, and allow defining it only via the provider config (tf). For example to enable debug we could do:

```
provider "auth0" {
    debug = "true"
}
```

I took this example from the aws provider, which only looks for certain env vars (that are usually defined in the host system) and for others you must configure it yourself.

What do you think?

Reply:
For aws, I think the core reason is that the provider just supports "well-known" environment variables, and no specific "provider" environment variables..

Personally, I think "debug" is more like a dynamic property than a static one. It would be convenient for me just using `export TF_LOG=debug` and `export AUTH0_DEBUG=true` if I want to debug auth0 requests temporary. If I would have to change tf file, I could end up accidentally committing this or something.

Still, if you don't think that it can be convenient, I can remove the support for environment variable.

Reply:
Yes, I see your point in that it would be a nuisance to change it in the file. Lets keep it then with env vars and see how it goes ;)